src/features/profile/components/EditPersonDialog/EditPersonFields.tsx

```diff
@@ -0,0 +1,240 @@
+import { DatePicker } from '@mui/x-date-pickers-pro';
+import { FC } from 'react';
+import UndoIcon from '@mui/icons-material/Undo';
+import {
+  Box,
+  FormControl,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Select,
+} from '@mui/material';
+import dayjs, { Dayjs } from 'dayjs';
+
+import EditPersonField from './EditPersonField';
+import formatUrl from 'utils/formatUrl';
+import globalMessageIds from 'core/i18n/globalMessageIds';
+import { makeNaiveDateString } from 'utils/dateUtils';
+import messageIds from 'zui/l10n/messageIds';
+import { NATIVE_PERSON_FIELDS } from 'features/views/components/types';
+import useCustomFields from '../../hooks/useCustomFields';
+import {
+  CUSTOM_FIELD_TYPE,
+  ZetkinCreatePerson,
+  ZetkinPerson,
+} from 'utils/types/zetkin';
+import { Msg, useMessages } from 'core/i18n';
+
+enum GENDERS {
+  FEMALE = 'f',
+  MALE = 'm',
+  OTHER = 'o',
+  UNKNOWN = 'unknown',
+}
+
+interface EditPersonFieldsProps {
+  fieldsToUpdate: ZetkinCreatePerson;
+  invalidFields: string[];
+  onChange: (field: string, newValue: string) => void;
+  onReset: (field: string) => void;
+  orgId: number;
+  fieldValues: ZetkinPerson;
+}
+
+const EditPersonFields: FC<EditPersonFieldsProps> = ({
+  fieldsToUpdate,
+  invalidFields,
+  onChange,
+  onReset,
+  orgId,
+  fieldValues,
+}) => {
+  const customFields = useCustomFields(orgId).data ?? [];
+  const globalMessages = useMessages(globalMessageIds);
+
+  return (
+    <Box display="flex" flexDirection="column" gap={2} paddingTop={1}>
+      <Box display="flex" gap={2}>
+        <EditPersonField
+          field={NATIVE_PERSON_FIELDS.FIRST_NAME}
+          hasChanges={NATIVE_PERSON_FIELDS.FIRST_NAME in fieldsToUpdate}
+          onChange={(field, newValue) => onChange(field, newValue)}
+          onReset={() => onReset(NATIVE_PERSON_FIELDS.FIRST_NAME)}
+          required
+          value={fieldValues.first_name}
+        />
+        <EditPersonField
+          field={NATIVE_PERSON_FIELDS.LAST_NAME}
+          hasChanges={NATIVE_PERSON_FIELDS.LAST_NAME in fieldsToUpdate}
+          onChange={(field, newValue) => onChange(field, newValue)}
+          onReset={() => onReset(NATIVE_PERSON_FIELDS.LAST_NAME)}
+          required
+          value={fieldValues.last_name}
+        />
```

Comment:
I was able to reset these to empty strings, and I don't think that should be allowed. Here are steps to reproduce:

1. Navigate to any person on the PR preview build ([example](https://app-zetkin-e0h9tgi1g-zetkin.vercel.app/organize/1/people/123))
2. Click the "EDIT" button
3. Clear the first name
4. Click SAVE

Expected: Saving should not be possible, prevented either by disabling the SAVE button or by responding with some message when saving.
Actual: The first name is set to the empty string, which is going to be allowed by the backend because it's non-null.

This needs to be solved on the frontend.

Reply:
Ooops